packages/next/src/server/use-cache/use-cache-wrapper.ts

```diff
@@ -73,10 +75,80 @@ cacheHandlerMap.set('default', {
   },
 })
 
-// TODO: Consider moving this another module that is guaranteed to be required in a safe scope.
-const runInCleanSnapshot = createSnapshot()
+function generateCacheEntry(
+  staticGenerationStore: StaticGenerationStore,
+  clientReferenceManifest: DeepReadonly<ClientReferenceManifest>,
+  cacheHandler: CacheHandler,
+  serializedCacheKey: string | ArrayBuffer,
+  encodedArguments: FormData | string,
+  fn: any
+): Promise<any> {
+  // We need to run this inside a clean AsyncLocalStorage snapshot so that the cache
+  // generation cannot read anything from the context we're currently executing which
+  // might include request specific things like cookies() inside a React.cache().
+  // Note: It is important that we await at least once before this because it lets us
+  // pop out of any stack specific contexts as well - aka "Sync" Local Storage.
```

Comment:
This sentence is copied over from the previous location where we called `runInCleanSnapshot`. Does it make sense here? We're not awaiting anything inside here, so should it be a note for the call site instead? We also still have the original comment in place. I'm not sure whether you intended to keep both?